felt/src/lib.rs

```diff
@@ -231,91 +848,83 @@ mod test {
         fn shift_left_in_range(ref value in "(0|[1-9][0-9]*)", ref shift_amount in "[0-9]{1,3}"){
             let value = Felt::parse_bytes(value.as_bytes(), 10).unwrap();
-            let p = &BigUint::parse_bytes(PRIME_STR[2..].as_bytes(), 16).unwrap();
             let shift_amount:u32 = shift_amount.parse::<u32>().unwrap();
             let result = (value << shift_amount).to_biguint();
-            prop_assert!(&result < p);
+            prop_assert!(&result <= &Felt::max_value().to_biguint());
         }
 
         #[test]
          // Property-based test that ensures, for 100 {value}s that are randomly generated each time tests are run, that performing a bit shift to the right by {shift_amount} of bits (between 0 and 999) returns a result that is inside of the range [0, p].
         fn shift_right_in_range(ref value in "(0|[1-9][0-9]*)", ref shift_amount in "[0-9]{1,3}"){
             let value = Felt::parse_bytes(value.as_bytes(), 10).unwrap();
-            let p = &BigUint::parse_bytes(PRIME_STR[2..].as_bytes(), 16).unwrap();
             let shift_amount:u32 = shift_amount.parse::<u32>().unwrap();
             let result = (value >> shift_amount).to_biguint();
-            prop_assert!(&result < p);
+            prop_assert!(&result <= &Felt::max_value().to_biguint());
         }
 
         #[test]
         // Property-based test that ensures, for 100 {value}s that are randomly generated each time tests are run, that performing a bit shift to the right by {shift_amount} of bits (between 0 and 999), with assignment, returns a result that is inside of the range [0, p].
         // "With assignment" means that the result of the operation is autommatically assigned to the variable value, replacing its previous content.
         fn shift_right_assign_in_range(ref value in "(0|[1-9][0-9]*)", ref shift_amount in "[0-9]{1,3}"){
             let mut value = Felt::parse_bytes(value.as_bytes(), 10).unwrap();
-            let p = FeltBigInt::parse_bytes(PRIME_STR[2..].as_bytes(), 16).unwrap();
             let shift_amount:usize = shift_amount.parse::<usize>().unwrap();
             value >>= shift_amount;
             value.to_biguint();
-            prop_assert!(value < p);
+            prop_assert!(value <= Felt::max_value());
         }
 
         #[test]
         // Property based test that ensures, for a pair of 100 values {x} and {y} generated at random each time tests are run, that performing a BitAnd operation between them returns a result that is inside of the range [0, p].
         fn bitand_in_range(ref x in "(0|[1-9][0-9]*)", ref y in "(0|[1-9][0-9]*)"){
             let x = Felt::parse_bytes(x.as_bytes(), 10).unwrap();
             let y = Felt::parse_bytes(y.as_bytes(), 10).unwrap();
-            let p = FeltBigInt::parse_bytes(PRIME_STR[2..].as_bytes(), 16).unwrap();
```

Comment:
Here's the bug. Compare `BigUint` with `BigUint` and everything will be OK.

Reply:
Resolved: https://github.com/lambdaclass/cairo-rs/pull/743/commits/c268a2bb81cd3e1e23a667895ab2ac8228e8979e